### PR TITLE
Bugfix/incorrect fallback

### PR DESCRIPTION
**Pull request by juampi92**
The idea of config(api.versions) is to keep an array of all the supported versions.

Whenever a version is deprecated, we remove them from the list.
Whenever a new version is added, we append it to the list (always at the end).

Having this progression can allow us to properly decide which version to use in case of a missing resource.

For example:
If we support v1, v2 and v3 (being v3 the latest), if we have the following resources: v2\User and v3\User, whenever we access v1, we should get v2 user, not the latest (v3).

To be able to go from v1 to v2, we need versions to be filled.

### CODE REVIEW

File: tests/ResourcePathResolveTest.php [hunk 1]
```diff
@@ -24,54 +25,54 @@ public function setUp(): void
...
         $this->assertEquals('\\App\\Http\\Resources\\App\\v1\\User', $classname);
 
         $this->apiResourceManager->setVersion('2');
-        $classname = $this->callMethod($this->apiResourceManager, 'parseClassname', ['App\Users']);
+        $classname = $this->callMethod($this->apiResourceManager, 'parseClassname', ['App\Users', '2']);
         $this->assertEquals('\\App\\Http\\Resources\\App\\v2\\Users', $classname);
 
         $this->apiResourceManager->setVersion('3');
-        $classname = $this->callMethod($this->apiResourceManager, 'parseClassname', ['App\User\Single']);
+        $classname = $this->callMethod($this->apiResourceManager, 'parseClassname', ['App\User\Single', '3']);
```
Comment (khalilst) [inline tests/ResourcePathResolveTest.php:36]: It sounds like we need some data providers in this file.

Comment (juampi92) [reply tests/ResourcePathResolveTest.php:36]: Yeah, we will delete this once Mazen finishes his refactor, to be honest

File: src/APIResourceManager.php [hunk 2]
```diff
@@ -66,36 +61,31 @@ public function getRouteName($route)
...
-     *  Get config considering the API name if present.
+     * Get config considering the API name if present.
+     *
+     * @param string $cfg Config path
+     * @param string $name Name of api if present
+     *
+     * @return mixed The result of the config
      */
-    protected function getConfig(string $key, string $apiName = null): ?string
+    protected function getConfig($cfg, $name = null)
```
Comment (sunchayn) [inline src/APIResourceManager.php:80]: shall we make the param more expressive? like: `configPath`

File: tests/ResourcePathResolveTest.php (hunk 1 above)
Comment (sunchayn) [reply tests/ResourcePathResolveTest.php:36]: This will be obsolete test yes

**Review by sunchayn — APPROVED** [review]
Nice

File: src/APIResourceManager.php (hunk 2 above)
Comment (juampi92) [reply src/APIResourceManager.php:80]: I'll be deleting this in the next PR

**Pull request merged**